src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -874,25 +923,21 @@ type DiagnosticItem =
                 "This expression may be short-circuited, and operation calls may not be executed."
             | WarningCode.DeprecationWithRedirect -> "{0} has been deprecated. Please use {1} instead."
             | WarningCode.DeprecationWithoutRedirect -> "{0} has been deprecated."
+
+            // Capability errors
             | WarningCode.UnsupportedResultComparison ->
-                "The target {0} does not support comparing measurement results."
+                DiagnosticItem.CapabilityErrorFormatString(ErrorCode.UnsupportedResultComparison)
```

Comment:
Consider
```suggestion
                DiagnosticItem.CapabilityErrorFormatString(WarningCode.UnsupportedResultComparison)
```

I have an impression that currently the `ErrorCode.UnsupportedResultComparison` and the `WarningCode.UnsupportedResultComparison` have the same value. But theoretically in the future they can get different values.

Reply:
The whole point of this block it to reuse the same string for errors and warnings.